india_compliance/gst_india/client_scripts/payment_entry.js

```diff
@@ -1,51 +1,147 @@
+const WARNING_ICON = `
+    <span class='warning-icon link-btn mb-auto mt-auto' style='display: block; z-index: 1; top: 5px; width: 24px; height: 24px'>
+        <div>${frappe.utils.icon('solid-warning', "md")}</div>
+    </span>
+`;
+
+
 frappe.ui.form.on("Payment Entry", {
-	company: function (frm) {
-		frappe.call({
-			method: "frappe.contacts.doctype.address.address.get_default_address",
-			args: {
-				doctype: "Company",
-				name: frm.doc.company,
-			},
-			callback: function (r) {
-				frm.set_value("company_address", r.message);
-			},
-		});
-	},
-
-	party: function (frm) {
-		update_gst_details(
-			frm,
-			"india_compliance.gst_india.overrides.payment_entry.update_party_details"
-		);
-	},
-
-	customer_address(frm) {
-		update_gst_details(frm);
-	},
+    setup(frm) {
+        override_get_outstanding_documents(frm);
+    },
+
+    refresh(frm) {
+        add_warning_indicator(frm, frm.doc.__onload?.reconciliation_status_dict);
+    },
+
+    company(frm) {
+        frappe.call({
+            method: "frappe.contacts.doctype.address.address.get_default_address",
+            args: {
+                doctype: "Company",
+                name: frm.doc.company,
+            },
+            callback(r) {
+                frm.set_value("company_address", r.message);
+            },
+        });
+    },
+
+    party(frm) {
+        update_gst_details(
+            frm,
+            "india_compliance.gst_india.overrides.payment_entry.update_party_details"
+        );
+    },
+
+    customer_address(frm) {
+        update_gst_details(frm);
+    },
 });
 
+function override_get_outstanding_documents(frm) {
+    const old_fn = frm.events?.get_outstanding_documents;
+    if (!old_fn) return;
+
+    const new_fn = function () {
+        old_fn(...arguments);
+        frappe.after_ajax(() => {
+
+            const invoice_list = frappe.last_response && frappe.last_response.message
+                ? frappe.last_response.message
+                    .filter(r => r.voucher_type === "Purchase Invoice")
+                    .map(r => r.voucher_no)
+                : [];
+
+            frappe.call({
+                method: "india_compliance.gst_india.overrides.payment_entry.get_reconciliation_status_for_invoice_list",
```

Comment:
How about using `override_whitelisted_method`? This is purely frontend functionality and should not break the backend.